crates/oxc_linter/src/config/config_builder.rs

```diff
@@ -149,7 +149,19 @@ impl ConfigStoreBuilder {
                                 ConfigStoreBuilder::from_oxlintrc(true, extended_config)?;
                             let rules = std::mem::take(&mut extended_config_store.rules);
                             builder = builder.with_rules(rules);
-                            builder = builder.and_plugins(extended_config_store.plugins(), true);
+
+                            // Handle plugin inheritance
+                            let parent_plugins = extended_config_store.plugins();
+                            let child_plugins = builder.plugins();
```

Comment:
We seem to add standard plugins by default when building this. 

So, below we need to check if child_plugins points to the same reference as `LintPlugins::default()` to infer whether the child config was initialized without `plugins` attribute defined.

I'm open to other ideas here!